.github/workflows/ci-check.yml

```diff
@@ -9,6 +9,7 @@ on:
     branches: [ master, citest ]
   pull_request:
     branches: [ master ]
+  workflow_dispatch:
```

Comment:
I think that's reasonnable given tests run pretty fast, so losing the parallel workflow execution would be ok for now

Reply:
These were intentionally split up since each test task itself takes up to 9 mins to execute. See a scan here: https://ge.solutions-team.gradle.com/s/ud7f5hnjz23zs/timeline?details=fhrjqjultghpo

Since test tasks in the same project do not run in parallel, this would lead to a 30min or so CI build.

Reply:
Actually I didn't pay attention some of the runs take great advantage of the build cache, so without caching they take much more (> 10 mins each on GH). I believe we should keep this parallelization with GH runners (they have only 2 cores, so not an option to run all tests in parallel on one single runner). Locally running all tests are already supported with `gradlew check`

Reply:
Then move the Android version definitions to GH instead of adding task paths to json files and reading them from GH actions.

Reply:
I'm not sure I entirely follow but if we move the version definitions purely to GH, it would make it a bit tricker to execute a test for a specific version of AGP locally. For example, I want to run the AGP 7.0 tests locally I would have to pass the system property that github does instead of just executing a named task. Do I understand this correctly?

Reply:
> Do I understand this correctly? Yes

Reply:
Since the "supported versions" is used elsewhere in the buid logic, I extracted those parts in a plugin (see latest commit). As part of this change the task now generates the json as part of stdout and captured as matrix settings by GH.

Reply:
What are we optimizing for here?
It seems to me that we are just moving the complexity from one place to another and making the local developer experience a bit more complicated.

Reply:
Resolving this, will create a new PR to move the conversation there